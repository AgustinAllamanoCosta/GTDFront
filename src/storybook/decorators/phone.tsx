import { GoogleOAuthProvider } from '@react-oauth/google';
import { styled } from 'styled-components';
import { Children } from 'react';

export const PhoneContext = (Story: any) => (
  <AppContext>
    <Story />
  </AppContext>
);

type AppContextProps = {
  children: string | JSX.Element | JSX.Element[];
};

export const AppContext = ({ children }: AppContextProps) => {
  return (
    <GoogleOAuthProvider clientId={process.env.CLEINT_ID ? process.env.CLEINT_ID: ''}>
      <MyPhoneContext>{children}</MyPhoneContext>
    </GoogleOAuthProvider>
  );
};

const MyPhoneContext = styled.div`
  background-color: black;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 390px;
  height: 884px;
`;
