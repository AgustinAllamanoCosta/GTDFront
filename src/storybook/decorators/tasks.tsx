import { v4 as uuidv4 } from 'uuid';
import { Task } from '../../types/types';
import { BrowserRouter } from 'react-router-dom';
import ItemsContext from '../../components/useItems/useItems';

const task1: Task = {
  title: 'some task 1',
  isComplete: false,
  id: uuidv4(),
  isCancele: false,
  isActive: true,
};
const task2: Task = {
  title: 'some task 2',
  isComplete: false,
  id: uuidv4(),
  isCancele: false,
  isActive: true,
};
const task3: Task = {
  title: 'some task 3',
  isComplete: false,
  id: uuidv4(),
  isCancele: false,
  isActive: true,
};

export const mockTaskInContext: Array<Task> = [task1, task2, task3];

export const TaskContextDecorator = (Story: any) => {
  return (
    <BrowserRouter>
      <ItemsContext>
        <Story />
      </ItemsContext>
    </BrowserRouter>
  );
};
