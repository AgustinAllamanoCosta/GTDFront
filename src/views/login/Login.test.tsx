import React from 'react';
import { expect } from '@jest/globals';
import { render, act, screen } from '@testing-library/react';
import LoginView from './Login';
import { AppContext } from '../../storybook/decorators/phone';

describe('Login View', () => {
  test('Should render a login view with singin and login buttons', async () => {
    const { container } = render(
      <AppContext>
        <LoginView />
      </AppContext>,
    );
    expect(container).toMatchSnapshot();
  });
});
