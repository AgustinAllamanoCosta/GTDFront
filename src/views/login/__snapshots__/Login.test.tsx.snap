// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login View Should render a login view with singin and login buttons 1`] = `
<div>
  <div
    class="sc-hdxXtv fSlFFO"
  >
    <div
      class="sc-dBwcjI ea-DaTq"
    >
      <div
        class="sc-kPNBzt fYtqUt"
        data-cy="Card-Header"
      >
        <div
          class="sc-jlOLRj bTdNNq"
        >
          <div
            class="sc-bXxmBy jXyjXk"
          >
            <span
              class="sc-dMnqAs dIjpAh"
              data-cy="Card-title"
            >
              Get Things Done
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-eErDDp bkdnWY"
        data-cy="Card-Contents"
      >
        <div
          class="sc-bWPEIA dhioSw"
        >
          <div
            class="sc-hfDuwB iNQNqE"
            data-cy="button-gtd"
          >
            <span
              class="sc-jCIOXC uLkTi"
              data-cy="button-text"
            >
              Login
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-address-card sc-fBEEOr eedbxR"
              data-cy="button-icon"
              data-icon="address-card"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm80 256h64c44.2 0 80 35.8 80 80c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16c0-44.2 35.8-80 80-80zm-32-96a64 64 0 1 1 128 0 64 64 0 1 1 -128 0zm256-32H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="sc-hfDuwB iNQNqE"
            data-cy="button-gtd"
          >
            <span
              class="sc-jCIOXC uLkTi"
              data-cy="button-text"
            >
              SingUp
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-plus sc-fBEEOr eedbxR"
              data-cy="button-icon"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
